src/attackmate/executors/browser/sessionstore.py

```diff
@@ -0,0 +1,159 @@
+import threading
+import time
+from playwright.sync_api import sync_playwright
+from queue import Queue, Empty
+
+
+class SessionThread(threading.Thread):
+    """
+    A dedicated thread that manages a single Playwright browser session (browser, context, page).
+    All commands for this session are queued to make sure they run synchronously in this thread.
+    """
+    def __init__(self, session_name=None, headless=False):
+        super().__init__()
+        self.session_name = session_name
+        self.headless = headless
+        self.cmd_queue = Queue()
+        self.res_queue = Queue()
+        self.playwright = None
+        self.browser = None
+        self.context = None
+        self.page = None
+        self.stopped = False
+        self.daemon = True  # mark this thread as a daemon so it won’t block interpreter exit
+        self.start()
+
+    def run(self):
+        """
+        Main loop of the thread. It:
+          1. Starts Playwright.
+          2. Waits for commands on the cmd_queue.
+          3. Executes them.
+          4. Puts results on the res_queue.
+          5. Shuts down on stop signal.
+        """
+        self.playwright = sync_playwright().start()
+
+        while not self.stopped:
+            try:
+                command = self.cmd_queue.get(timeout=0.1)
+            except Empty:
+                continue
+
+            if command is None:
+                break  # signals that we're done
+
+            cmd_name, args, kwargs = command
+            try:
+                result = self._handle_command(cmd_name, *args, **kwargs)
+                self.res_queue.put((False, result))
+            except Exception as e:
+                self.res_queue.put((True, str(e)))
```

Comment:
this is not essential, but nicer for debugging instead of str(e) you could put the error on the res_queue and re-raise it in submit_command to preserve traceback. i.e. have things on the queue like (result, none) or (none, error)

Reply:
@thorinaboenke and @annaerdi is this issue resolved? I'd like to merge